lib/shared/layers/python-sdk/python/genai_core/types.py

```diff
@@ -31,6 +31,8 @@ class Provider(Enum):
     BEDROCK = "bedrock"
     OPENAI = "openai"
     SAGEMAKER = "sagemaker"
+    AMAZON = "amazon"
+    COHERE = "cohere"
```

Comment:
This is confusing semantic as Bedrock and Sagemaker are not Provider in the same sense as Amazon and Cohere are. Please split Model Provider from Hosting Providers.

Reply:
I agree, it's not clean. I'll split it and push today.

Reply:
Splitting the providers into `ModelProviders` and `HostingProviders` has a way bigger impact than I expected. `Provider `is widely used and `ModelProvider` is oddly used in the react-app/common/types. At this point I've made changes in 20 files but the sprawl of changes needed continues. 

Do you want me to continue to do the split or am I just approaching this incorrectly and overcomplicating it?

Reply:
@QuinnGT while I agree there should be a separation the give the ramification of this split I'd say is out of the scope of this PR. So if you want to revert back to just provider we can create a separate issue for this refactor.

Reply:
@bigadsoleiman okay, I didn't commit my changes so it should all be stable as is now. Fork is sync'd with main now. 

You can approve the PR and close it.